perception/autoware_map_based_prediction/src/map_based_prediction_node.cpp

```diff
@@ -360,17 +361,11 @@ CrosswalkEdgePoints getCrosswalkEdgePoints(const lanelet::ConstLanelet & crosswa
 }
 
 bool withinRoadLanelet(
-  const TrackedObject & object, const lanelet::LaneletMapPtr & lanelet_map_ptr,
+  const TrackedObject & object,
+  const std::vector<std::pair<double, lanelet::Lanelet>> & surrounding_lanelets_with_dist,
   const bool use_yaw_information = false)
 {
-  const auto & obj_pos = object.kinematics.pose_with_covariance.pose.position;
-  lanelet::BasicPoint2d search_point(obj_pos.x, obj_pos.y);
-  // nearest lanelet
-  constexpr double search_radius = 10.0;  // [m]
-  const auto surrounding_lanelets =
-    lanelet::geometry::findNearest(lanelet_map_ptr->laneletLayer, search_point, search_radius);
-
-  for (const auto & lanelet_with_dist : surrounding_lanelets) {
+  for (const auto & lanelet_with_dist : surrounding_lanelets_with_dist) {
     const auto & dist = lanelet_with_dist.first;
```

Comment:
Can you
```suggestion
for (const auto & [dist, lanelet] : surrounding_lanelets_with_dist)
```
for clarity ?

Reply:
Thanks!! Modified.